apps/explorer/src/ui/CopyToClipboard.tsx

```diff
@@ -28,31 +34,65 @@ const iconStyles = cva(['cursor-pointer hover:text-steel-dark'], {
 
 export type IconStylesProps = VariantProps<typeof iconStyles>;
 
-export interface CopyToClipboardProps extends IconStylesProps {
+export interface CopyToClipboardProps extends Omit<IconStylesProps, 'success'> {
 	copyText: string;
 	onSuccessMessage?: string;
 }
 
-const ICON_SIZES = {
+const COPY_ICON_SIZES = {
 	sm: Copy12,
 	md: Copy16,
 	lg: Copy24,
 };
 
+const CHECK_ICON_SIZES = {
+	sm: Check12,
+	md: CheckStroke16,
+	lg: CheckStroke24,
+};
+
+const TIMEOUT_TIMER = 2000;
+
 export function CopyToClipboard({
 	copyText,
 	color,
 	size = 'md',
 	onSuccessMessage = 'Copied!',
 }: CopyToClipboardProps) {
+	const [showCopyIcon, setShowCopyIcon] = useState(true);
 	const copyToClipBoard = useCopyToClipboard(() => toast.success(onSuccessMessage));
 
-	const CopyIcon = ICON_SIZES[size!];
+	const CopyIcon = COPY_ICON_SIZES[size!];
+	const CheckIcon = CHECK_ICON_SIZES[size!];
+
+	const onClickHandler = async () => {
+		await copyToClipBoard(copyText);
+		setShowCopyIcon(false);
+	};
+
+	useEffect(() => {
+		if (!showCopyIcon) {
+			const timeout = setTimeout(() => {
+				setShowCopyIcon(true);
+			}, TIMEOUT_TIMER);
+			return () => clearTimeout(timeout);
+		}
+	}, [showCopyIcon]);
 
 	return (
-		<Link onClick={() => copyToClipBoard(copyText)}>
+		<Link disabled={!showCopyIcon} onClick={onClickHandler}>
 			<span className="sr-only">Copy</span>
-			<CopyIcon className={iconStyles({ size, color })} />
+			{showCopyIcon ? (
+				<motion.div
+					initial={{ opacity: 0 }}
+					animate={{ opacity: 1 }}
+					transition={{ duration: 0.1 }}
```

Comment:
this duration is so short that the effect isn't noticeable

Reply:
Change to 0.2s